plugins/infinite-sky.cpp

```diff
@@ -0,0 +1,270 @@
+
+#include "Core.h"
+#include "DataDefs.h"
+#include "Debug.h"
+#include "Export.h"
+#include "LuaTools.h"
+#include "PluginManager.h"
+
+#include "modules/World.h"
+#include "modules/EventManager.h"
+
+#include "df/construction.h"
+#include "df/map_block.h"
+#include "df/map_block_column.h"
+#include "df/world.h"
+#include "df/z_level_flags.h"
+
+#include <cstring>
+#include <string>
+#include <vector>
+
+using namespace std;
+
+using namespace DFHack;
+using namespace df::enums;
+
+DFHACK_PLUGIN("infinite-sky");
+DFHACK_PLUGIN_IS_ENABLED(is_enabled);
+
+REQUIRE_GLOBAL(world);
+
+namespace DFHack {
+    // for configuration-related logging
+    DBG_DECLARE(infiniteSky, control, DebugCategory::LINFO);
+    // for logging during creation of z-levels
+    DBG_DECLARE(infiniteSky, cycle, DebugCategory::LINFO);
+}
+
+static const string CONFIG_KEY = string(plugin_name) + "/config";
+static PersistentDataItem config;
+enum ConfigValues {
+    CONFIG_IS_ENABLED = 0,
+};
+
+command_result infiniteSky (color_ostream &out, std::vector <std::string> & parameters);
+
+static void constructionEventHandler(color_ostream& out, void* ptr);
+EventManager::EventHandler handler(plugin_self, constructionEventHandler,0);
```

Comment:
0 might be a bit too aggressive since it would cause the EventManager to do a linear scan on every tick. How about 11?